funsor/ops/array.py

```diff
@@ -66,20 +66,7 @@ def _logaddexp(x, y):
 sample = SampleOp(_logaddexp, name="sample")
 
 
-class ReshapeMeta(type):
-    _cache = {}
-
-    def __call__(cls, shape):
-        shape = tuple(shape)
```

Comment:
I think we'll still need this line to be able to jit PyTorch functions.

Reply:
Ok, I put this back into a simpler `ReshapeMeta`. When we refactor `function`, we may want to move this logic inside of `CachedOpMeta` and always convert any args of `torch.Size` to tuples.